jtec-core/src/main/java/edu/tum/sse/jtec/reporting/LCOVReportParser.java

```diff
@@ -0,0 +1,194 @@
+        Map<String, Path> classToSourceFile = new HashMap<>();
+        for (Path sourceFile : sourceFiles) {
+            final String sourceCode = IOUtils.readFromFile(sourceFile);
+            final CompilationUnit compilationUnit = StaticJavaParser.parse(sourceCode);
+
+            List<String> fullyQualifiedClassNames = compilationUnit
+                    .findAll(ClassOrInterfaceDeclaration.class).stream()
+                    .map(ClassOrInterfaceDeclaration::getFullyQualifiedName)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collect(Collectors.toList());
+            for (String className : fullyQualifiedClassNames)
+                classToSourceFile.put(className, sourceFile);
+        }
+        return classToSourceFile;
+    }
+
+    /**
+     * For each method in each source file, create an LCOV section with hit count 0 for all lines.
+     * @return The list of initialized LCOV sections
+     */
+    private List<LCOVSection> getInitializedLcovSections() throws IOException {
+        List<LCOVSection> lcovSections = new ArrayList<>();
+        for (Path sourceFile : sourceFiles) {
+            final LCOVSection lcovSection = new LCOVSection("", sourceFile.toString());
+            final String sourceCode = IOUtils.readFromFile(sourceFile);
+            final CompilationUnit compilationUnit = StaticJavaParser.parse(sourceCode);
+
+            compilationUnit.findAll(CallableDeclaration.class)
+                    .forEach(callableDeclaration -> {
+                        final Range range = callableDeclaration.getRange().orElse(null);
+                        if (range == null)
+                            return;
+                        lcovSection.addMethod(range.begin.line, callableDeclaration.getNameAsString());
+                        for (int line = range.begin.line; line <= range.end.line; line++)
+                            lcovSection.addLineHit(line, 0);
+                    }
+            );
+            lcovSections.add(lcovSection);
+        }
+        return lcovSections;
+    }
+
+    /**
+     * Get the Java source file corresponding to a class
+     * @param className The fully qualified class name
+     * @return The source file if it exists, else null
+     */
+    private Path getJavaSourceFile(String className) {
+        return classToSourceFile.get(className);
+    }
+
+    /**
+     * Extract the package name, class name, and method name from an entity string.
+     * If no method name is present, use null for the method.
+     * @return An array [package, class, method], or null if no match is found
+     */
+    public static String[] getEntityComponents(String entity) {
+        Matcher matcher = entityRegex.matcher(entity);
+        if (matcher.matches()) {
+            MatchResult result = matcher.toMatchResult();
+            return new String[] {result.group(1), result.group(2), result.group(3)};
+        }
+        return null;
```

Comment:
hm, not so sure what I think about the pattern of returning `null` here if no match is found. it requires the caller to always check for non-null return values. I personally would prefer `Optional<String[]>` as it makes it explicit that the return value has to be checked. You could simply return `Optional.empty()` in the default case. What do you think?

Reply:
Sounds good, replaced the `null` checks with `Optional`